cityofaustin/atd-moped
# Paginate user request and suppress sign up emails

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/8532
https://github.com/cityofaustin/atd-data-tech/issues/8507

this PR suppresses the email AWS triggers when a new user is added to cognito. 

It also fixes a bug where the request to cognito for our list of users was limited to 60. Cognito provides a paginator to request all of the users. 

## Testing
**URL to test:** 


**Steps to test:**
Get your JWT token from moped staging, and put that in the header as your Authorization token (Authorization: Bearer <JWT token>) in Postman/Insomnia etc. 

Request https://moped-api-test.austinmobility.io/users and compare the difference to the response you get from https://moped-api-staging.austinmobility.io/users. When I tested, my changes returned 42 users where staging was only returning 31. (The endpoint filters out azure account duplicates). 


OR spin up local moped-api instance. 
Request all the users: 127.0.0.1:5000/users. 

Running moped locally, add a new user. (This will add the user to the staging cognito pool and staging database). You should not get an email welcoming you to moped. You can then delete the user from the db and the cognito pool. 


---
#### Ship list
